backend/lib/resolve-payload-cids.js

```diff
@@ -106,3 +117,31 @@ async function updatePayloadCidInActiveDeal (pgPool, deal, newPayloadRetrievalSt
     throw Error(util.format('Error updating payload of deal: ', deal), { cause: error })
   }
 }
+
+function getSmartContractClient () {
+  const fetchRequest = new ethers.FetchRequest(RPC_URL)
+  assert(GLIF_TOKEN, 'GLIF_TOKEN is required')
+  fetchRequest.setHeader('Authorization', `Bearer ${GLIF_TOKEN}`)
+  const provider = new ethers.JsonRpcProvider(fetchRequest)
+  return new ethers.Contract(
+    MINER_TO_PEERID_CONTRACT_ADDRESS,
+    MINER_TO_PEERID_CONTRACT_ABI,
+    provider
+  )
+}
+const defaultSmartContractClient = getSmartContractClient()
+
+/**
+ * @param {number} minerId
+ * @param {object} options
+ * @param {unknown} options.smartContract
+ * @param {import('./typings.js').MakeRpcRequest} options.makeRpcRequest
+ * @returns {Promise<{ peerId: string, source: string }>}
+ */
+export const getPeerId = async (minerId, { smartContract, makeRpcRequest } = { smartContract: defaultSmartContractClient, makeRpcRequest: rpcRequest }) => {
+  return await getIndexProviderPeerId(
+  `f0${minerId}`,
+  smartContract,
+  { rpcFn: makeRpcRequest }
+  )
+}
```

Comment:
We should add a timeout for this call, similarly to what we had in in the original implementation in spark-checker.

https://github.com/CheckerNetwork/spark-checker/pull/129#discussion_r2026312602

It seems to me we didn't configure any timeout in the original implementation in spark-deal-observer. Feel free to create a follow-up issue (or a follow-up pull request) if you prefer to add the timeout after this PR is landed.